src/app.js

```diff
@@ -7,6 +7,17 @@ export function app(app) {
   var node
   var element
 
+  function hydrate(elm) {
+    return elm == null ? undefined : {
+      tag: elm.tagName,
+      data: {},
+      children: [].map.call(elm.childNodes, function(child){hydrate(child)})
+    }
+  }
+
+  element = root.firstChild
+  node = hydrate(element)
```

Comment:
it's recursive so it would need to be function or it would need a loop and a stack to accomplish, there is discussion about moving this logic upto the SSR code and turning this into:
```js
node = JSON.parse(document.getElementById('nodes'))
```
Which would mean in the SSR code we would build the vnode object and serialize it to a script tag we send down from server like:
```html
<html>
<head>
  <script type='application/json'>
  {
    tag: 'div',
    data: {},
    children: [...]
  }
  </script>
</head>
</html>
```
This came up yesterday in the following discussion on hyperapp/server: https://github.com/hyperapp/server/issues/7